minifiers/terser.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+const fingerprinter = require('fingerprinting');
+const Config = require('./config.js');
+const Terser = require('terser');
+
+/**
+ * Minify JS using Terser.
+ *
+ * @param {Object} options Content (compiled JS) and filepath.
+ * @param {Function} done Completion callback.
+ * @returns {Undefined} Return early on minification error.
+ * @public
+ */
+module.exports = function terser(options, done) {
+  const filename = options.file.replace('.js', '.min.js');
+  const config = new Config({
+    minify: options.minify,
+    sourceMapContent: options.map,
+    filename
+  });
+
+  const results = Terser.minify(options.content.toString('utf-8'), config.terser);
+
+  //
+  // Return on a minification error.
+  //
+  if (results.error) {
+    return done(results.error);
```

Comment:
while we're in here, can we make sure we know when minification is starting/done in status events (and probably which minifier is being used)? and make sure that this error from the minifiers get tagged in status events in such a way that we know they originated from minification (and which minfiier).

Reply:
Good idea, but I'm going to consider this out of scope. https://github.com/godaddy/carpenterd-worker/blob/master/builder.js#L2 `workers-factory` is only a dependency of `carpenterd-worker` so I assume we most likely need to either do the work there or give the `workers-factory` access to the `writeStream`. I don't consider that to be something we want though. Ideally `workers-factory` emits events which we can listen to somewhere around here https://github.com/godaddy/carpenterd-worker/blob/master/builder.js#L251-L272 Happy to make a ticket for this though.